pages/api/xAPI/statements/index.ts

```diff
@@ -51,92 +46,93 @@ async function runMiddleware(
 ): void {
   return new Promise((resolve, reject) => {
     fn(req, res, (result) => {
-      if( result instanceof Error) {
-        return reject(result)
+      if (result instanceof Error) {
+        return reject(result);
       }
 
       return resolve(result);
-    })
-  })
+    });
+  });
 }
 
 /******************************GET, HEAD***************************************/
 async function handleGET(req: NextRequest, res: NextResponse): void {
-  const client = new MongoClient(uri, options);
   try {
     // await runMiddleware(req, res, sanitizeQueryParams);
     // await runMiddleware(req, res, validateQueryParams);
     // await runMiddleware(req, res, cors);
-    await client.connect();
+    await dbClient.connect();
     const queryOptions = generateQueryParams(req.query);
-    const rows = await client
+    const rows = await dbClient
       .db("xAPI")
       .collection("statements")
       .find(queryOptions.params)
       .sort(queryOptions.sort)
       .toArray();
     rows.forEach((row) => {
-      delete row['_id']; //_id is a mongoDB parameter outside the scope of xAPI
+      delete row["_id"]; //_id is a mongoDB parameter outside the scope of xAPI
     });
     res.status(200).json({
       statements: rows,
-      more: ''
-    })
+      more: "",
+    });
   } catch (err) {
     throw err;
-  } finally{
-    client.close();
+  } finally {
+    await dbClient.close();
```

Comment:
Added `await` to prevent the method from _returning_ while executing `.close()`. 

In other words, having `await dbClient.close();` _ensures_ `.close()` completes before `handleGET()` returns.